Backend/app/routes/auth.py

```
@@ -1,7 +1,60 @@
+# Pydantic schemas for data validation
+class UserCreate(BaseModel):
+    username: str
+    email: EmailStr
+    password: str
+
+class UserLogin(BaseModel):
+    email: EmailStr
+    password: str
+
+@router.post("/signup")
+async def signup(user_data: UserCreate, db: AsyncSession = Depends(get_db)):
```

Ruff (0.15.1)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)